tests/core/test_shots.py

```diff
@@ -0,0 +1,76 @@
+from datetime import datetime, timezone
+from lagom import Container
+
+from parlant.core.agents import Agent
+from parlant.core.customers import Customer
+from parlant.core.engines.alpha.guideline_proposer import (
+    GuidelineProposer,
+    GuidelinePropositionSchema,
+    GuidelinePropositionShot,
+    GuidelinePropositionsSchema,
+)
+from parlant.core.guidelines import GuidelineContent
+from parlant.core.sessions import Event, EventId
+from parlant.core.shots import ShotCollection
+
+
+async def test_that_appended_shot_is_displayed_in_guideline_proposer_prompt(
+    container: Container,
+    agent: Agent,
+    customer: Customer,
+) -> None:
+    guideline_proposer = container[GuidelineProposer]
+    shot_collection = container[ShotCollection[GuidelinePropositionShot]]
+
+    guideline_content = GuidelineContent(
+        condition="A user compliments the product",
+        action="Tell him to stop being a bitch",
```

Comment:
Ehh maybe dial it down a bit :P

Reply:
haha I wasn't sure what you'll think about it